### PR TITLE
propagate Postgres table comments to GraphQL schema descriptions (close #446)

### LINKED ISSUE #446: propagate comments from sql to graphql schema
(via ben on cherre)

@0x777 I remember you saying that there was a problem with this. What was it?
cc @rakeshkky

### PR DESCRIPTION
**Pull request by ajeetdsouza**
### Description
The console currently allows a user to add a description to a table via `Data >  Schema > schema_name > table_name > Modify`, but it does not show this description anywhere else. This PR adds descriptions to the GraphQL schema, viewable via the Documentation Explorer:
image

### Affected components 
- Server

### Related Issues
#446 

### Solution and Design
Descriptions are stored via [Postgres comments](https://www.postgresql.org/docs/current/sql-comment.html), viewable in the `pg_descriptions` table. A `descriptions` field has been added to the `hdb_catalog.hdb_table_info_agg` view, and this description is concatenated with the default description when being served to the frontend.

### Steps to test and verify
Add a description to any of the tables as described above. It should now be visible in the Documentation Explorer.

### Limitations, known bugs & workarounds
None

### To the Reviewer (server)
- No new modules have been added.
- The catalog version has been upgraded to `18`. A migration for this has been added (`from17To18`), which drops the `hdb_catalog.hdb_table_info_agg` view and recreates it with an added `description` field.
- A newtype `PGDescription` has been added.
- `TableInfo` now has a new field for descriptions: `tiDescription :: Maybe PGDescription`.
- The following functions now take an extra parameter for table descriptions: `mkTableObj`, `mkGCtxRole`, `mkGCtxRole'`.

**Critical files:**
- `src-exec/Migrate.hs`
- `src-lib/Hasura/SQL/Types.hs`
- `src-lib/Hasura/GraphQL/Schema.hs`
- `src-lib/Hasura/RQL/Types/SchemaCache.hs`
- `src-rsr/initialise.sql`
- `src-rsr/migrate_from_17_to_18.sql`

### CODE REVIEW

File: server/src-lib/Hasura/RQL/Types/SchemaCache.hs [hunk 1]
```diff
@@ -343,6 +343,7 @@ mutableView qt f mVI operation =
 data TableInfo
   = TableInfo
   { tiName                  :: !QualifiedTable
+  , tiDescription           :: !(Maybe Text)
```
Comment (rakeshkky) [inline server/src-lib/Hasura/RQL/Types/SchemaCache.hs]: `newtype` for descriptions. `PGDescription`?

File: server/src-lib/Hasura/RQL/Types/SchemaCache.hs [hunk 2]
```diff
@@ -357,14 +358,15 @@ $(deriveToJSON (aesonDrop 2 snakeCase) ''TableInfo)
 instance FromJSON TableInfo where
   parseJSON = withObject "TableInfo" $ \o -> do
     name <- o .: "name"
+    description <- o .:? "description"
```
Comment (rakeshkky) [inline server/src-lib/Hasura/RQL/Types/SchemaCache.hs]: convey `"description"` is a `Maybe` value

**Comment by rakeshkky** [comment]
@ajeetdsouza 
Can a table have multiple descriptions? If yes, how are you handling them?

**Comment by ajeetdsouza** [comment]
@rakeshkky
Postgres allows only one comment per object, so this won't be a problem: https://www.postgresql.org/docs/current/sql-comment.html

**Comment by coco98** [comment]
@ajeetdsouza Why aren't we propagating other comments? Let's just propagate all of them?
I tried column comments, but it doesn't work.

In order of priority, I would say:
1. Tables
2. Views
3. Columns
4. Functions

**Comment by coco98** [comment]
I don't think this is correct. 
I added comments to table/column. 
https://hge-ci-pull-2397.herokuapp.com/console/data/schema/public/tables/test/modify

They show up in the wrong place on the graphql docs explorer

**Comment by mastoj** [comment]
I still don't believe this to be fixed. I went ahead and created a table and a column on the heroku instance. I can't see the comment I added for the column when I browse through docs.

**Comment by rakeshkky** [comment]
> I still don't believe this to be fixed. I went ahead and created a table and a column on the heroku instance. I can't see the comment I added for the column when I browse through docs.

@mastoj Thank you reporting this. This has been fixed in the next commit.

**Comment by mastoj** [comment]
@rakeshkky nice. Verified that it works.

**Comment by rakeshkky** [comment]
@lexi-lambda Ready to be merged.

File: server/src-rsr/migrate_from_22_to_23.sql [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  rtn.nspname::text AS return_type_schema,
+  rt.typname::text AS return_type_name,
+
+  CASE
+    WHEN ((rt.typtype) :: text = ('b' :: character(1)) :: text) THEN 'BASE' :: text
+    WHEN ((rt.typtype) :: text = ('c' :: character(1)) :: text) THEN 'COMPOSITE' :: text
+    WHEN ((rt.typtype) :: text = ('d' :: character(1)) :: text) THEN 'DOMAIN' :: text
+    WHEN ((rt.typtype) :: text = ('e' :: character(1)) :: text) THEN 'ENUM' :: text
+    WHEN ((rt.typtype) :: text = ('r' :: character(1)) :: text) THEN 'RANGE' :: text
+    WHEN ((rt.typtype) :: text = ('p' :: character(1)) :: text) THEN 'PSUEDO' :: text
```
Comment (lorenzo) [inline server/src-rsr/migrate_from_22_to_23.sql]: Is `PSUEDO` a typo? should it be `PSEUDO`?

Comment (rakeshkky) [reply server/src-rsr/migrate_from_22_to_23.sql]: @lorenzo Thank for pointing it out. Yeah, it's a typo.

**Pull request merged**